Benchmarks/GroupByVersusForLoopDictionary.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+
+namespace Benchmarks.Benchmarks {
+	/// <summary>
+	/// Question: How long does it take to turn an enumerable into a dictionary key'd off of one of the enumerable's properties
+	/// Methods: 
+	///		ForLoopDictionary - Have a dictionary, do lookup/insert of the key in a for loop
+	///		GroupBy (Linq) - Run GroupBy, then ToDictionary linq methods
+	/// Data: All keys are ints
+	///		DistinctList - No repeated keys in list
+	///		AllSameKey - Only 1 repeated key in list
+	///		100Keys - 100 possible keys, repeated via modulo throughout the source list
+	/// Results: Forloop dictionary took about half the time of the groupby method and was slightly more memory efficient
```

Comment:
I probably wouldn't include results because overtime they may change with optimizations in the framework - at least for frameworks still being worked on.

Reply:
Fair, was hoping for an at-a-glance explanation without having to rerun but will remove. Framework changes (or even differences, since we all know I didn't run all of them) will affect it.